samples/pd/ambix_read~.c

```diff
@@ -606,8 +606,8 @@ static t_int *ambix_read_perform(t_int *w) {
 
     /* check for EOF (and buffer is about to drain) */
     if (x->x_eof &&
-        x->x_fifohead > x->x_fifotail &&
-        x->x_fifohead <=  x->x_fifotail + wantframes-1
+        x->x_fifohead >= x->x_fifotail &&
+        x->x_fifohead < x->x_fifotail + wantframes-1
         ) {
```

Comment:
On 17/04/16 22:28, umläute wrote:
>
> In samples/pd/ambix_read~.c 
> <https://github.com/umlaeute/ambix/pull/1#discussion_r59991881>:
>
> > @@ -606,8 +606,8 @@ static t_int *ambix_read_perform(t_int *w) {
> >
> >      /* check for EOF (and buffer is about to drain) */
> >      if (x->x_eof &&
> > -        x->x_fifohead > x->x_fifotail &&
> > -        x->x_fifohead <=  x->x_fifotail + wantframes-1
> > +        x->x_fifohead >= x->x_fifotail &&
> > +        x->x_fifohead < x->x_fifotail + wantframes-1
> >          ) {
>
> what was wrong here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/umlaeute/ambix/pull/1/files/9557794d1b2405bc562f45f33082c370a8167b39#r59991881>
>
this change fixes a bug where the playback is not stopped at the end of 
file.
(instead it gets looped, which is not the wanted behavior)